### PR TITLE
retrievers: add merger_retriever

### PR DESCRIPTION
**Pull request by Abirdcfly**
For kubeagi/arcadia#925

### CODE REVIEW

**Comment by Abirdcfly** [comment]
wait, i need fix golangci-lint error

File: retrievers/merger_retriever.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+	}
+}
+
+func (m *MergerRetriever) GetRelevantDocuments(ctx context.Context, query string) ([]schema.Document, error) {
+	if m.CallbacksHandler != nil {
+		m.CallbacksHandler.HandleRetrieverStart(ctx, query)
+	}
+	docs := make([]schema.Document, 0)
+	for _, r := range m.Retrievers {
+		doc, err := r.GetRelevantDocuments(ctx, query)
```
Comment (nkwangleiGIT) [inline retrievers/merger_retriever.go:34]: can we do this in parallel for better performance?

File: retrievers/merger_retriever.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+var _ schema.Retriever = &MergerRetriever{}
+
+// MergerRetriever is a retriever that merges the results of multiple retrievers.
+type MergerRetriever struct {
+	Retrievers       []schema.Retriever
+	CallbacksHandler callbacks.Handler
+}
+
+// NewMergerRetriever creates a new MergerRetriever.
+func NewMergerRetriever(
```
Comment (nkwangleiGIT) [inline retrievers/merger_retriever.go:19]: maybe it's better to use MergeRetriever for the name

Comment (Abirdcfly) [reply retrievers/merger_retriever.go:19]: We're consistent with the naming of langchain python...they call it Merger Retriever in doc and mergerretiever in code.

File: retrievers/merger_retriever.go (hunk 1 above)
Comment (bjwswang) [reply retrievers/merger_retriever.go:34]: Parallel may increase the overhead of embedding service,so it's better to add another parameter to controll this.  we can do this in another pr.

**Review by bjwswang — APPROVED** [review]
let's have a try.

**Pull request merged**